graph/src/ipfs_client.rs

```diff
@@ -12,6 +14,37 @@ use std::fmt::Display;
 use std::time::Duration;
 use std::{str::FromStr, sync::Arc};
 
+#[derive(Debug, thiserror::Error)]
+pub enum IpfsError {
+    #[error("Request error: {0}")]
+    Request(#[from] reqwest::Error),
+    #[error("IPFS file {0} is too large. It can be at most {1} bytes")]
+    FileTooLarge(String, usize),
+}
+
+impl IpfsError {
+    pub fn is_timeout(&self) -> bool {
+        match self {
+            Self::Request(e) => e.is_timeout(),
+            _ => false,
+        }
+    }
+
+    pub fn is_status(&self) -> bool {
```

Comment:
What does `is_status` indicate? Would be good to have a comment on what that means